js/adapt-scrollPrompt.js

```diff
@@ -1,25 +1,28 @@
-define([
-  'core/js/adapt',
-  './scrollPromptView'
-], function(Adapt, ScrollPromptView) {
+import Adapt from 'core/js/adapt';
+import ScrollPromptView from './ScrollPromptView';
 
-  Adapt.on('menuView:ready pageView:ready', function(view) {
+class ScrollPrompt extends Backbone.Controller {
 
-    var model = view.model;
+  initialize() {
+    this.listenTo(Adapt, {
+      'menuView:ready pageView:ready blockView:postRender componentView:postRender': this.setupView
+    });
+  }
 
-    var scrollPrompt = model.get('_scrollPrompt');
-    if (!scrollPrompt || !scrollPrompt._isEnabled) return;
+  setupView(view) {
+    const model = view.model;
 
-    var modelType = model.get('_type');
-    if (modelType === 'course') modelType = 'menu';
+    const scrollPrompt = model.get('_scrollPrompt');
+    if (!scrollPrompt || !scrollPrompt._isEnabled) return;
 
-    /* set model type selector to append scroll prompt to */
-    var modelTypeSelector = '.' + modelType + '__header-inner';
+    const modelType = model.get('_type');
+    const selector = modelType === 'component' ? '.component__inner' : `.${modelType}__header-inner`;
```

Comment:
Should block level render to `.block__inner`? Otherwise we have the same issue we previously had with component where the button doesn't render if no display text is set (no block header is rendered).

Reply:
@kirsty-hames Thanks for catching that. Here's what this looks like now:

![block-level-inner](https://github.com/cgkineo/adapt-scrollPrompt/assets/898168/08fc72e2-f22c-4c1a-8685-b1d78b28afaf)